### ISSUE TITLE
Validation label appears in a training batch

### ISSUE THREAD

**Issue by btseytlin** [KevinMusgrave/powerful-benchmarker#75]
I think I found a potential crippling bug.

During training, in my custom loss function, I print this:
```
def compute_loss(self, embeddings, labels, indices_tuple):
  uniq_labels = sorted(np.array(torch.unique(labels).cpu()))
  print('Unique labels according to batch', uniq_labels)
```
Result:
```
Unique labels according to batch [3, 15, 24, 27, 50, 60, 63, 70]
```

But according to `split_manager.get_label_set("eval", "train")`, train labels start at 25, labels 3, 15, and 24 are actually from the validation labels!

Hope there is some mistake on my side, otherwise, it would appear that we were training our models on validation all along!
Am I not understanding how it works? Maybe the loss function doesn't get labels, but instead gets indices of training labels or something?

Details:

I run the following experiment:
```
python3 runners/global_embedding_regularizer.py --experiment_name global_embedding_regularizer \
--dataset~OVERRIDE~ {CUB200: {download: True}} \
--bayes_opt_iters 25 \
--factories {hook~OVERRIDE~: {CustomHookFactory: {}}} \
--loss_funcs~OVERRIDE~ \
{metric_loss: {ContrastiveLossRegularized: {\
reg_own_weight~BAYESIAN~: [0, 1], reg_own_threshold~BAYESIAN~: [0, 1], \
reg_other_weight~BAYESIAN~: [0, 1], reg_other_threshold~BAYESIAN~: [0, 1], \
reg_warmup~INT_BAYESIAN~: [0, 10], \
pos_margin~BAYESIAN~: [0, 1], \
neg_margin~BAYESIAN~: [0, 1]}}} 
```

I have a custom loss and a custom hook factory that provides an end of epoch hook.

In the end of epoch hook I print the label sets for train, test and val:
```
def end_of_epoch_hook(trainer):

            print('Train label set', split_manager.get_label_set("eval", "train"))
            print('Test label set', split_manager.get_label_set("eval", "test"))
            print('Val label set', split_manager.get_label_set("eval", "val"))
   
           # Irrelevant code further on
```

And in my loss function I print the labels in batch as above

**Comment by KevinMusgrave** [comment]
It's because ```set_min_label_to_zero``` is set to ```True``` in the trainer. So no matter what the true validation labels are, the loss function will see them as ranging from 0 to N-1, where N is the number of classes in the validation set. The purpose is to make the labels work with classification losses.

**Comment by btseytlin** [comment]
Thank you, now my problem is resolved
